### ISSUE TITLE
AuditScope.Create(... Ignores ForEntity<T>.Ignore() and [AuditIgnore]

### ISSUE THREAD

**Issue by jaredfabus-careform** [thepirat000/Audit.NET#193]
I'm using the following libraries from Audit.Net:
Audit.EntityFramework
Audit.NET
Audit.NET.SqlServer

This is for an MVC App using EF6

I'm not sure if this isn't working as intended or perhaps I haven't configured it correctly or am missing some configuration but here's what I'm seeing:

On MVC App start in Global.asax.cs I'm configuring auditing in the following way:

`Audit.EntityFramework.Configuration.Setup()
                .ForAnyContext(config => config
                    .IncludeEntityObjects(false)
                    .IncludeIndependantAssociations(false)
                    .ForEntity<TestAudit>(_ => _
                        .Ignore(audit => audit.FirstName)
                        .Ignore(audit => audit.LastName)))
                .UseOptIn();`

My DbContext definition:
`public partial class MyEntities : AuditDbContext`

TestAudit's object definition doesn't have any audit decorations or anything so it just looks like this:
`public partial class TestAudit
    {
        public int SystemId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Disease { get; set; }
    }`

Finally, here's the implementation of the auditing:
`using (var db = Db.Factory.Create())
            {
                var auditRecord = db.TestAudits.FirstOrDefault();
                
                using (AuditScope.Create("TestAudits:Update", () => auditRecord))
                {
                    auditRecord.FirstName = "Firsty" + new Random().Next().ToString();
                    auditRecord.LastName = "Lasty" + new Random().Next().ToString();
                    auditRecord.Disease = "Chickenpox" + new Random().Next().ToString();

                    db.SaveChanges();
                }
            }`

I've puposefully used OptIn and not decorated the TestAudit object with [AuditInclude] because I only want an audit log generated with my scope, NOT every time a TestAudit object is modified.  With that in mind, despite configuring to ignore FirstName and LastName, my log looks like this using the default data provider:
`"Target": {
    "Type": "TestAudit",
    "Old": {
      "SystemId": 1,
      "FirstName": "Firsty1275996953",
      "LastName": "Lasty1275996953",
      "Disease": "Chickenpox1275996953"
    },
    "New": {
      "SystemId": 1,
      "FirstName": "Firsty672258970",
      "LastName": "Lasty672258970",
      "Disease": "Chickenpox672258970"
    }
  },`

Am I wrong in thinking that AuditScope.Create(... should be paying attention to the configured .Ignore( definitions from the Fluent Setup functions?
Perhaps I've configured it wrong?

Any help is appreciated.

Thanks.

**Comment by thepirat000** [comment]
When using the extensions that generates events, like the `Audit.EntityFramework`, you don't have to explicitly create the `AuditScope`/`AuditEvent`. They will be created internally by the extension, you only need to worry about the configuration of the `DbContext` and the configuration of the Audit output.

For example for `Audit.EntityFramework`, you could proceed as follows:

- Change your `DbContext` to inherit from `AuditDbContext`
- Configure the `Audit.EntityFramework` via `Audit.EntityFramework.Configuration.Setup().` 
- Configure the `Audit.NET` output via `Audit.Core.Configuration.Setup().`

So in your code you should remove the `using(AuditScope.Create...` and just modify the entities and call `SaveChanges`. Note that only one audit event is generated for each `SaveChanges` call, even if there are multiple modifications.

I don't know where do you want to store the audit logs, I guess on a SQL server, so an example of the output configuration could be:

```
Audit.Core.Configuration.Setup()
    .UseSqlServer(config => config
        .ConnectionString("data source=localhost;initial catalog=Audit;integrated security=true;")
	.Schema("dbo")
        .TableName("Event")
        .IdColumnName("EventId")
        .JsonColumnName("Data")
        .LastUpdatedColumnName("LastUpdatedDate"));
```

**Comment by thepirat000** [comment]
You're right, that's not the intended purpose. If you want to ignore a property on the JSON serialization you have some options, please check this [SO answer](https://stackoverflow.com/a/54086213/122195) for some examples.
